src/main/scala/io/smartdatalake/workflow/DAG.scala

```diff
@@ -267,15 +268,18 @@ object DAG extends SmartDataLakeLogger {
    * Create a lookup table to retrieve outgoing (target) node IDs for a node.
    */
   private def buildOutgoingIdLookupTable(nodes: Seq[DAGNode], edges: Seq[DAGEdge]) = {
-    val targetIDsforIncommingIDsMap = edges.groupBy(_.nodeIdFrom).mapValues(_.map(_.nodeIdTo))
-    nodes.map(n => (n, targetIDsforIncommingIDsMap.getOrElse(n.nodeId, Seq()))).toMap
+    val targetIDsforIncomingIDsMap = edges.groupBy(_.nodeIdFrom).mapValues(_.map(_.nodeIdTo))
+    nodes.map(n => (n, targetIDsforIncomingIDsMap.getOrElse(n.nodeId, Seq()))).toMap
   }
 
   /**
    * Create a lookup table to retrieve incoming (source) node IDs for a node.
    */
   private def buildIncomingIdLookupTable(nodes: Seq[DAGNode], edges: Seq[DAGEdge]) = {
-    val incomingIDsForTargetIDMap = edges.groupBy(_.nodeIdTo).mapValues(_.map(_.nodeIdFrom))
+    //val incomingIDsForTargetIDMap = edges.groupBy(_.nodeIdTo).mapValues(_.map(_.nodeIdFrom))
```

Comment:
H @swissrussdev , easier would be 
`val incomingIDsForTargetIDMap = edges.groupBy(_.nodeIdTo).mapValues(_.map(_.nodeIdFrom).distinct)
`

Reply:
Ok, i pushed the change. I'll merge then for release 1.0 :-)